### PR TITLE
refactor!(crypto): Don't process in-room verification implicitly

### PR DESCRIPTION
**Pull request by poljar**
This closes: #1341.

### CODE REVIEW

File: bindings/matrix-sdk-crypto-ffi/src/machine.rs [hunk 1]
```diff
@@ -827,6 +827,25 @@ impl OlmMachine {
         Ok(())
     }
 
+    /// Receive an verification event.
```
Comment (jplatte) [inline bindings/matrix-sdk-crypto-ffi/src/machine.rs:830]:
```suggestion
    /// Receive a verification event.
```

File: crates/matrix-sdk-crypto/src/machine.rs [hunk 2]
```diff
@@ -731,13 +730,22 @@ impl OlmMachine {
...
     /// those will get passed to the `OlmMachine` during decryption.
+    #[deprecated(note = "Use OlmMachine::receive_verification_event instead", since = "0.7")]
     pub async fn receive_unencrypted_verification_event(
         &self,
         event: &AnyMessageLikeEvent,
     ) -> StoreResult<()> {
         self.verification_machine.receive_any_event(event).await
     }
 
+    /// Receive an verification event.
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/machine.rs:741]:
```suggestion
    /// Receive a verification event.
```

File: bindings/matrix-sdk-crypto-ffi/src/olm.udl [hunk 3]
```diff
@@ -396,6 +396,7 @@ interface OlmMachine {
 
     [Throws=CryptoStoreError]
     void receive_unencrypted_verification_event([ByRef] string event, [ByRef] string room_id);
+    void receive_verification_event([ByRef] string event, [ByRef] string room_id);
```
Comment (BillCarsonFr) [inline bindings/matrix-sdk-crypto-ffi/src/olm.udl:399]: [Throws=CryptoStoreError]

File: crates/matrix-sdk-crypto/src/machine.rs [hunk 4]
```diff
@@ -731,13 +730,22 @@ impl OlmMachine {
...
+    ///
+    /// This method can be used to pass verification events that are happening
+    /// in rooms to the `OlmMachine`.
```
Comment (BillCarsonFr) [inline crates/matrix-sdk-crypto/src/machine.rs:744]: Maybe clarify that it should be a decrypted event (or clear event)? 
What about the `m.relates_to` field? it's not encrypted, should it be copy back in the resulting decrypted content?

Comment (poljar) [reply crates/matrix-sdk-crypto/src/machine.rs:744]: Hmm, would you know the type of the event if it was encrypted? Presumably no, but yeah, we can spell that out.

**Review by BillCarsonFr — APPROVED** [review]
Tested on ER and working. Fixing my issues with verification events replay.

**Pull request merged**